src/dense/generic_dense.jl

```diff
@@ -580,23 +602,108 @@ function ode_interpolant(Θ, dt, y₀, y₁, k, cache::OrdinaryDiffEqMutableCach
+    end
+end
+
+function partial_hermite_interpolant(Θ, dt, y₀::AbstractArray, y₁::AbstractArray, k, cache, idxs::Int, T::Type{Val{TI}}, differential_vars) where {TI}
+    if differential_vars[idxs]
+        hermite_interpolant(Θ, dt, y₀, y₁, k, Val{cache isa OrdinaryDiffEqMutableCache}, idxs, T)
+    else
+        linear_interpolant(Θ, dt, y₀, y₁, idxs, T)
+    end
+end
+
+function partial_hermite_interpolant(Θ, dt, y₀::Number, y₁::Number, k, cache, idxs, T::Type{Val{TI}}, differential_vars) where {TI}
+    if only(differential_vars)
+        hermite_interpolant(Θ, dt, y₀, y₁, k, Val{cache isa OrdinaryDiffEqMutableCache}, idxs, T)
+    else
+        linear_interpolant(Θ, dt, y₀, y₁, idxs, T)
+    end
+end
+
+function partial_hermite_interpolant(Θ, dt, y₀, y₁, k, cache, idxs, T::Type{Val{TI}}, differential_vars) where {TI}
+    throw("how did we get here")
+    @assert all(differential_vars)
+    hermite_interpolant(Θ, dt, y₀, y₁, k, Val{cache isa OrdinaryDiffEqMutableCache}, idxs, T)
+end
+
+function partial_hermite_interpolant!(out::AbstractArray, Θ, dt, y₀::AbstractArray, y₁::AbstractArray, k, cache, idxs::AbstractVector{Int}, T::Type{Val{TI}}, differential_vars) where {TI}
+    sel = differential_vars[idxs]
+    diffidxs = idxs[sel]
+    linidxs = idxs[.!sel]
+    hermite_interpolant!(@view(out[sel]), Θ, dt, y₀, y₁, k, diffidxs, T)
+    linear_interpolant!(@view(out[.!sel]), Θ, dt, y₀, y₁, linidxs, T)
+    out
+end
+
+function partial_hermite_interpolant!(out::AbstractArray, Θ, dt, y₀::AT, y₁::AT, k, cache, idxs::Nothing, T::Type{Val{TI}}, differential_vars) where {TI, AT <: AbstractArray}
+    if ArrayInterface.fast_scalar_indexing(AT)
+        partial_hermite_interpolant!(out, Θ, dt, y₀, y₁, k, cache, eachindex(IndexLinear(), y₀), T, differential_vars)
+    else
+        h = hermite_interpolant(Θ, dt, y₀, y₁, k, Val{cache isa OrdinaryDiffEqMutableCache}, idxs, T)
+        l = linear_interpolant(Θ, dt, y₀, y₁, idxs, T)
+        @.. broadcast=false out=h*differential_vars + l*!differential_vars
+    end
+end
+
+function partial_hermite_interpolant!(out::AbstractArray, Θ, dt, y₀::AbstractArray, y₁::AbstractArray, k, cache, idxs::Int, T::Type{Val{TI}}, differential_vars) where {TI}
+    if differential_vars[idxs]
+        hermite_interpolant!(out, Θ, dt, y₀, y₁, k, idxs, T)
+    else
+        linear_interpolant!(out, Θ, dt, y₀, y₁, idxs, T)
+    end
+end
+
+function partial_hermite_interpolant!(out, Θ, dt, y₀::Number, y₁::Number, k, cache, idxs, T::Type{Val{TI}}, differential_vars) where {TI}
+    if only(differential_vars)
+        hermite_interpolant!(out, Θ, dt, y₀, y₁, k, idxs, T)
+    else
+        linear_interpolant!(out, Θ, dt, y₀, y₁, idxs, T)
+    end
 end
 
-function _ode_interpolant!(out, Θ, dt, y₀, y₁, k, cache, idxs, T::Type{Val{TI}}) where {TI}
+function partial_hermite_interpolant!(out, Θ, dt, y₀, y₁, k, cache, idxs, T::Type{Val{TI}}, differential_vars) where {TI}
+    throw("how did we get here")
```

Comment:
Once I'm confident I've handled all the cases I guess either completely remove this method, or remove the error and leave it as a fallback if it turns out there is some insane edge case somewhere in the ecosystem.